ts/RootContainer.tsx

```diff
@@ -34,13 +65,52 @@ const styles = StyleSheet.create({
  */
 class RootContainer extends React.Component<Props> {
   public componentDidMount() {
-    AppState.addEventListener("change", this.onApplicationActivityChange);
+    if (Platform.OS === "android") {
+      Linking.getInitialURL()
+        .then(this.navigate)
+        .catch(console.error);
+    } else {
+      Linking.addEventListener("url", this.handleOpenURL);
+    }
+
+    AppState.addEventListener("change", this.handleApplicationActivityChange);
   }
 
   public componentWillUnmount() {
-    AppState.removeEventListener("change", this.onApplicationActivityChange);
+    Linking.removeEventListener("url", this.handleOpenURL);
+    AppState.removeEventListener(
+      "change",
+      this.handleApplicationActivityChange
+    );
   }
 
+  private handleOpenURL = (event: { url: string }) => {
+    this.navigate(event.url);
+  };
+
+  private navigate = (url: string | null) => {
+    if (!url) {
+      return;
+    }
+
+    const route = url.slice(ROUTES.PREFIX.length);
+    const routeParts = route.split("/");
+    const routeName = routeParts[0];
+    const id = routeParts[1] || undefined;
+
+    const deeplinkNavigationPayload = { routeName, id };
+
+    if (this.props.appState === "background") {
```

Comment:
add a short comment explaining the two behaviors, I have no idea what `setDeeplink` and `navigateToDeeplink` do